### ISSUE TITLE
[Bug]: Not able to fetch data from the Google sheet with size of 7.5 MB 

### ISSUE THREAD

**Issue by harshit98** [appsmithorg/appsmith#27628]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Trying to fetch data from a google sheet with size of 7.5 mb but it gives me exception of `DataBufferLimitException` with limit exceeding 10485760 bytes i.e 10.4 mb.

Digging deeper into the issue - I found that the spring boot framework has by default 10 mb data buffer limit: https://stackoverflow.com/questions/59735951/databufferlimitexception-exceeded-limit-on-max-bytes-to-buffer-webflux-error

But, somehow I am confused as well that why 7.5 mb file is facing data fetch issue if the limit is upto 10 mb max. Adding screenshots of the steps I followed and the error I got while fetching data from the Google sheet.

I'm following the steps mentioned in this video - https://community.appsmith.com/tutorial/building-crud-app-google-sheets

![image (4)](https://github.com/appsmithorg/appsmith/assets/22245418/d2289d21-b6c0-4cec-9fd4-887acac2df76)
![image (3)](https://github.com/appsmithorg/appsmith/assets/22245418/72a573b1-cf8a-422f-b546-a89a5dbbe67c)

### Steps To Reproduce

1. Go to Datasources.
2. Add data source of Google sheet.
3. Give authorisation of the Google sheet (read, write, delete)

### Environment

Production

### Version

v1.9.19

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_1tauh8)

**Comment by harshit98** [comment]
Also, it would be great to know that how can I upload large size Google sheets on the Appsmith? 

My usecase is something different where I want to show data in form of visualisations from the sheet. The sheet size would probably will be more than 10 mb defined limit.

**Comment by harshit98** [comment]
@Nikhil-Nandagopal @pranavkanade 

Any update on this issue? I'm currently looking for a quick help to solve this issue at my end.

**Comment by Nikhil-Nandagopal** [comment]
@harshit98 I don't understand what you mean by upload large sheet because it seems you are using a fetch query to return data from a google sheet. Am I right in understanding that you are not able to fetch the data?

**Comment by harshit98** [comment]
Yes @Nikhil-Nandagopal, I'm trying to fetch this data from a google sheet.

**Comment by ame-appsmith** [comment]
I have been able to reproduce this bug on version v1.9.37.1. Use case:

The user tries fetching data from a Google Sheet with a huge amount of rows and only 7 columns. Although the **Pagination Limit** is set to `10` in the query, when running it, the error is: `Your query failed to execute:
org.springframework.core.io.buffer.DataBufferLimitException: Exceeded limit on max bytes to buffer : 10485760`.
Additionally, I tried to fetch only one column or filter the sheet using a WHERE clause so that only 1 record would be returned, but I'm getting the same error.

[Link to screen recording](https://www.loom.com/share/eb793a45beba4a35895e9137040b2357?sid=6dc60b92-5ed3-41cb-8cfe-7960b504c2f6)
[Link to sample app shared with Appsmith Support](https://app.appsmith.com/app/gs-databufferlimitexception/page1-6515912506f6133585858764/edit/saas/google-sheets-plugin/api/6515918adfc56b535d2c6e06)

More details and sample data can be found in this Discord thread: https://discord.com/channels/725602949748752515/1155810910464659536

**Comment by harshit98** [comment]
@ame-appsmith 

Cool, thanks! So, if this is a bug - is it fixed in the latest releases of Appsmith or will it be taken up by Engineering team?

**Comment by ame-appsmith** [comment]
@harshit98 This bug is still open; it is still present in the latest release (v1.9.37.1)

**Comment by harshit98** [comment]
@ame-appsmith - Got it. What's the tentative date for the bug fix?

**Comment by geekyme-fsmk** [comment]
I have the same problem here with a large sheet. It's 6.1MB, with 144749 rows.

**Comment by karlfrankeser** [comment]
I have the same issues here:
```
"sheets": [
    {
      "sheetId": 0,
      "title": "Sheet1",
      "index": 0,
      "sheetType": "GRID",
      "gridProperties": {
        "rowCount": 28592,
        "columnCount": 26
      }
    },
    {
      "sheetId": 829241355,
      "title": "Data",
      "index": 1,
      "sheetType": "GRID",
      "gridProperties": {
        "rowCount": 9182,
        "columnCount": 26
      }
    }
  ]
```

Running on version v1.9.61
